querydsl-docs/src/main/docbook/en-US/content/general/codegen.xml

```diff
@@ -242,6 +242,116 @@ import com.querydsl.core.annotations.Config;
 </project>
 ]]></programlisting>
 
+    <para>Alternatively <code>maven-compiler-plugin</code> can be used as well:
```

Comment:
Yes. It does exactly what you thought it would: runs APT processing during compilation. It also adds the generated sources folder to project sources, as generated classes are also compiled on the same pass (also IntelliJ Idea can see the generated sources folder).